File: samples/client/petstore/java/retrofit/src/main/java/io/swagger/client/CollectionFormats.java
package io.swagger.client;

import java.util.Arrays;
import java.util.List;

public class CollectionFormats {
    
    public static class CSVParams {

        protected List<String> params;
        
        public CSVParams() {
        }
        
        public CSVParams(List<String> params) {
            this.params = params;
        }
        
        public CSVParams(String... params) {
            this.params = Arrays.asList(params);
        }

        public List<String> getParams() {
            return params;
        }

        public void setParams(List<String> params) {
            this.params = params;
        }
        
        @Override
        public String toString() {
            return String.join(",", params);
        }
        
    }
    
    public static class SSVParams extends CSVParams {
        
        public SSVParams() {
        }
        
        public SSVParams(List<String> params) {
            super(params);
        }

        public SSVParams(String... params) {
            super(params);
        }

        @Override
        public String toString() {
            return String.join(" ", params);
        }
    }
    
    public static class TSVParams extends CSVParams {
        
        public TSVParams() {
        }
        
        public TSVParams(List<String> params) {
            super(params);
        }
        
        public TSVParams(String... params) {
            super(params);
        }

        @Override
        public String toString() {
            return String.join("\t", params);
        }
    }
    
    public static class PIPESParams extends CSVParams {
        
        public PIPESParams() {
        }
        
        public PIPESParams(List<String> params) {
            super(params);
        }
        
        public PIPESParams(String... params) {
            super(params);
        }

        @Override
        public String toString() {
            return String.join("|", params);
        }
    }
    
}

File: samples/client/others/kotlin-oneOf-anyOf-kotlinx-serialization-explicitApi/src/main/kotlin/org/openapitools/client/infrastructure/CollectionFormats.kt
package org.openapitools.client.infrastructure

public class CollectionFormats {

    public open class CSVParams {

        public var params: List<String>

        public constructor(params: List<String>) {
            this.params = params
        }

        public constructor(vararg params: String) {
            this.params = listOf(*params)
        }

        override fun toString(): String {
            return params.joinToString(",")
        }
    }

    public open class SSVParams : CSVParams {

        public constructor(params: List<String>) : super(params)

        public constructor(vararg params: String) : super(*params)

        override fun toString(): String {
            return params.joinToString(" ")
        }
    }

    public class TSVParams : CSVParams {

        public constructor(params: List<String>) : super(params)

        public constructor(vararg params: String) : super(*params)

        override fun toString(): String {
            return params.joinToString("\t")
        }
    }

    public class PIPESParams : CSVParams {

        public constructor(params: List<String>) : super(params)

        public constructor(vararg params: String) : super(*params)

        override fun toString(): String {
            return params.joinToString("|")
        }
    }

    public class SPACEParams : SSVParams()
}
